lib/src/client.dart

```diff
@@ -2760,25 +2760,30 @@ class Client extends MatrixApi {
         continue;
       }
 
-      var update =
-          EventUpdate(roomID: room.id, type: type, content: event.toJson());
-      if (event.type == EventTypes.Encrypted && encryptionEnabled) {
-        update = await update.decrypt(room);
+      if (event is MatrixEvent &&
+          event.type == EventTypes.Encrypted &&
+          encryptionEnabled) {
+        final decrypted = await encryption?.decryptRoomEvent(
+          Event.fromMatrixEvent(event, room),
+          updateType: type,
+        );
 
         // if the event failed to decrypt, add it to the queue
```

Comment:
nitpick but this comment should be right above the else case

Reply:
Fixed :)